src/base/emu-i386/simx86/cpatch.c

```diff
@@ -176,6 +176,28 @@ asmlinkage void rep_movs_stos(struct rep_stack *stack)
 			else repstos(,l,);
 		}
 	}
+	else if ((op & 0xf6) == 0xa6) { /* cmps/scas */
+		int repmod = (size == 1 ? MBYTE : size == 2 ? DATA16 : 0);
+		AR1.pu = stack->edi;
+		TR1.d = stack->ecx;
+		repmod |= MOVSDST|MREPCOND|(eip[-1]==REPNE? MREPNE:MREP);
+		if ((op & 0xf6) == 0xa6) { /* cmps */
+			repmod |= MOVSSRC;
+			AR2.pu = stack->esi;
+			Gen_sim(O_MOVS_CmpD, repmod);
+			stack->esi = AR2.pu;
+		}
+		else { /* scas */
+			DR1.d = stack->eax;
+			Gen_sim(O_MOVS_ScaD, repmod);
+		}
+		FlagSync_All();
+		stack->edi = AR1.pu;
+		ecx = TR1.d;
```

Comment:
Why one via stack-> and another one directly?
I.e. I see why it works, but perhaps something
to clean up?

Reply:
the `stack->ecx = ecx` is at the bottom of this function.
It could be cleaned up a bit indeed (also to eliminate that "goto").
Shall I add a commit to this PR or a new PR?

Reply:
To this one, or even better, just squash into
the same patch. :) To not touch the same place
in multiple patches.

Reply:
Or another patch if you mean some more
global clean-up.

Reply:
a more global cleanup is a bit more involving so I just did the two line change and merged it into the one patch,